include/experimental/__p2642_bits/layout_padded.hpp

```diff
@@ -0,0 +1,808 @@
+        "extents_type::static_extent(extent-to-pad) is 0 or dynamic_extent");
+    return __find_next_multiple(_PaddingValue,
+                                _ExtentsType::static_extent(_ExtentToPadIdx));
+  } else {
+    return dynamic_extent;
+  }
+}
+
+template <size_t _PaddingValue, typename _Extents, size_t _ExtentToPadIdx, size_t _Rank, typename Enabled = void>
+struct __static_array_type_for_padded_extent
+{
+  static constexpr size_t __padding_value = _PaddingValue;
+  using __index_type = typename _Extents::index_type;
+  using __extents_type = _Extents;
+  using __type = ::MDSPAN_IMPL_STANDARD_NAMESPACE::detail::maybe_static_array<
+      __index_type, size_t, dynamic_extent,
+      detail::__get_actual_static_padding_value<__extents_type, __padding_value,
+                                                _ExtentToPadIdx>()>;
+};
+
+template <size_t _PaddingValue, typename _Extents, size_t _ExtentToPadIdx, size_t Rank>
+struct __static_array_type_for_padded_extent<_PaddingValue, _Extents,
+                                             _ExtentToPadIdx, Rank, std::enable_if_t<Rank <= 1>> {
+  using __index_type = typename _Extents::index_type;
+  using __extents_type = _Extents;
+  using __type =
+      ::MDSPAN_IMPL_STANDARD_NAMESPACE::detail::maybe_static_array<
+          __index_type, size_t, dynamic_extent, 0>;
+};
+
+template <size_t _PaddingValue, typename _Extents, size_t _ExtentToPadIdx>
+struct __padded_extent {
+  static constexpr size_t __padding_value = _PaddingValue;
+  using __index_type = typename _Extents::index_type;
+  using __extents_type = _Extents;
+  using __static_array_type = typename __static_array_type_for_padded_extent<
+      __padding_value, _Extents, _ExtentToPadIdx, _Extents::rank()>::__type;
+
+  static constexpr auto static_value() { return __static_array_type::static_value(0); }
+
+  MDSPAN_INLINE_FUNCTION
+  static constexpr __static_array_type
+  __impl_init_padding(const _Extents &__extents) {
+    if constexpr ((_Extents::rank() > 1) && (__padding_value == dynamic_extent)) {
+      return {__extents.extent(_ExtentToPadIdx)};
+    } else {
+      return __impl_init_padding(__extents, __padding_value);
+    }
+  }
+
+  MDSPAN_INLINE_FUNCTION static constexpr __static_array_type
+      __impl_init_padding(const _Extents &__extents,
+                          __index_type __padding_value) {
+    if constexpr (_Extents::rank() > 1) {
+      return {__find_next_multiple(__padding_value,
+                                   __extents.extent(_ExtentToPadIdx))};
+    } else {
+      return {};
+    }
+  }
+
+  template <typename _Mapping, size_t _PaddingStrideIdx>
+  MDSPAN_INLINE_FUNCTION static constexpr __static_array_type
+  __impl_init_padding(const _Mapping &__other_mapping,
+                      std::integral_constant<size_t, _PaddingStrideIdx>) {
+    if constexpr (_Extents::rank() > 1) {
+      return {__other_mapping.stride(_PaddingStrideIdx)};
+    } else {
+      return {};
+    }
+  }
+};
+}
```

Comment:
```suggestion
} // namespace detail
```
We don't enforce this via clang-format?!